src/main/scala/rules/SymbolicExecutionRules.scala

```diff
@@ -29,14 +29,20 @@ trait SymbolicExecutionRules extends Immutable {
       val model = v.decider.getModel()
       if (model != null && !model.contains("model is not available")){
         val nativeModel = Model(model)
-        var ce: Counterexample = if (Verifier.config.counterexample.toOption.get == "native") {
-          val oldHeap = if (s.oldHeaps.contains(Verifier.PRE_STATE_LABEL))
-            Some(s.oldHeaps(Verifier.PRE_STATE_LABEL).values)
-          else None
-          SiliconNativeCounterexample(s.g, s.h.values, oldHeap, nativeModel)
-        }else{
-          SiliconVariableCounterexample(s.g, nativeModel)
+        val ce_type = Verifier.config.counterexample.toOption.get
+        var ce: Counterexample = ce_type match {
+          case "native" => 
+            val oldHeap = if (s.oldHeaps.contains(Verifier.PRE_STATE_LABEL))

```

Comment:
`contains` followed by getting the contained value is potentially inefficient (though the difference is most likely irrelevant here).

Better: Call `getOrElse` on `s.oldHeaps`, or get an `Option` from `s.oldHeaps` and use its `fold`

Reply:
I tried this but can't get it to work because of some typing problems. But this is not really my code anyways, it's still from the original native counterexamples so I hope it's okay if I leave it as it is.

Reply:
Please try again; the original source of the code doesn't really matter. Let me know if you can't get it to work, then I'll have a go at it.

Reply:
Okay I worked it out using Option.empty instead of None.